test/plasmamvp/deposits.js

```diff
@@ -0,0 +1,224 @@
+
+        let blockNum = (await instance.lastCommittedBlock.call()).toNumber() + 1;
+        await instance.submitBlock([toHex(root)], [1], [0], blockNum, {from: authority});
+
+        // create the confirm sig
+        let confirmHash = sha256String(merkleHash + root.slice(2));
+        let confirmSig = await web3.eth.sign(accounts[2], confirmHash);
+
+        // start the malicious exit
+        await instance.startDepositExit(nonce, 0, {from: accounts[2], value: minExitBond});
+
+        // checks matching inputs
+        let err;
+        [err] = await catchError(instance.challengeExit([0,0,0,nonce-1], [blockNum, 0, 0],
+            toHex(txBytes), toHex(sigs), toHex(proof), toHex(confirmSig), {from: accounts[3]}));
+        if (!err)
+            assert.fail("did not check against matching inputs");
+
+        // correctly challenge
+        await instance.challengeExit([0,0,0,nonce], [blockNum, 0, 0],
+            toHex(txBytes), toHex(proof), toHex(confirmSig), {from: accounts[3]});
+
+        let balance = (await instance.balanceOf.call(accounts[3])).toNumber();
+        assert.equal(balance, minExitBond, "challenger not awarded exit bond");
+
+        let exit = await instance.depositExits.call(nonce);
+        assert.equal(exit[4], 2, "exit state not changed to challenged");
+
+        // Cannot challenge twice
+        [err] = await catchError(instance.challengeExit([0,0,0,nonce], [blockNum, 0, 0],
+            toHex(txBytes), toHex(sigs), toHex(proof), toHex(confirmSig), {from: accounts[3]}));
+        if (!err)
+            assert.fail("Allowed a challenge for an exit already challenged");
+    });
+
+    it("Allows operator to challenge a deposit spend committing to an incorrect fee", async () => {
+        let nonce = (await instance.depositNonce.call()).toNumber();
+        await instance.deposit(accounts[2], {from: accounts[2], value: 100});
+
+        let txList = Array(17).fill(0);
+        txList[4] = nonce; txList[16] = 5; // fee
+        txList[12] = accounts[1]; txList[13] = 100;
+        let txHash = web3.sha3(RLP.encode(txList).toString('hex'), {encoding: 'hex'});
+        let sigs = [toHex(await web3.eth.sign(accounts[2], txHash)), toHex(Buffer.alloc(65).toString('hex'))];
+
+        let txBytes = [txList, sigs];
+        txBytes = RLP.encode(txBytes).toString('hex');
+
+        // submit the block
+        let merkleHash = sha256String(txBytes);
+        let [merkleRoot, proof] = generateMerkleRootAndProof([merkleHash], 0);
+        let blockNum = (await instance.lastCommittedBlock.call()).toNumber() + 1;
+        await instance.submitBlock([toHex(merkleRoot)], [1], [5], blockNum, {from: authority});
+
+        // exit the deposit not committing to the fee
+        await instance.startDepositExit(nonce, 1, {from: accounts[2], value: minExitBond});
+
+        // challenge the exit
+        await instance.challengeFeeMismatch([0,0,0,nonce], [blockNum, 0, 0], toHex(txBytes), toHex(proof));
```

Comment:
can you add an err check here to ensure that it is successfully challenged

Reply:
Or a check that startDepositExit doesn't fail

Reply:
added the check for the challenge. the deposit will revert if it doesn't succeed.